compiler/testData/codegen/bytecodeText/inlineClasses/resultApiDoesntUseBox.kt

```diff
@@ -1,3 +1,4 @@
+// IGNORE_BACKEND: JVM
```

Comment:
Maybe it's better to extract the failing part of this test into a separate file and file a YouTrack issue, since we don't have a habit of looking for tests ignored for _JVM_ backend and fixing those

Reply:
Also, after changing the behavior for Result, consider adding a separate test checking that we don't invoke `Result.equals-impl0` in the bytecode (that test will change its behavior in 1.4)

Reply:
Done